src/Plutarch/Extra/Rational.hs

```diff
@@ -0,0 +1,122 @@
+{-# LANGUAGE PolyKinds #-}
+-- TODO: Remove this and replace with TermCont.
+{-# LANGUAGE QualifiedDo #-}
+{-# LANGUAGE RankNTypes #-}
+-- TODO: Either disable warning about orphans altogether or address the issue
+-- requiring this unusual solution.
+{-# OPTIONS_GHC -Wwarn=orphans #-}
+
+module Plutarch.Extra.Rational (
+    mulTruncate,
+    mulDivTruncate,
+    divTruncate,
+    mulRational,
+    divRational,
+    pliftTaggedRational,
+) where
+
+-------------------------------------------------------------------------------
+
+import Data.Maybe (fromJust)
+import Data.Tagged (Tagged)
+import GHC.Stack (HasCallStack)
+import Plutarch.Builtin (pforgetData)
+import Plutarch.Extra.Tagged (PTagged)
+import qualified Plutarch.Monadic as P
+import Plutarch.Orphans ()
+import Plutarch.Positive (ptryPositive)
+import PlutusTx (fromData)
+
+--------------------------------------------------------------------------------
+
+{- | Combined multiply-truncate
+
+ @since 3.8.0
+-}
+mulTruncate ::
+    forall (s :: S).
+    Term s (PRational :--> PInteger :--> PInteger)
+mulTruncate =
+    phoistAcyclic $
+        plam $ \ex x -> P.do
+            (PRational num denom) <- pmatch ex
+            mulDivTruncate # x # num # pto denom
+
+{- | Multiply the first argument by the second argument, divide by the third,
+ truncating
+
+ @since 3.8.0
+-}
+mulDivTruncate ::
+    forall (s :: S).
+    Term s (PInteger :--> PInteger :--> PInteger :--> PInteger)
+mulDivTruncate =
+    phoistAcyclic $
+        plam $ \x num denom -> P.do
+            pdiv # (num * x) # denom
+
+{- | Combined divide-truncate
+
+ @since 3.8.0
+-}
+divTruncate ::
+    forall (s :: S).
+    Term s (PRational :--> PInteger :--> PInteger)
+divTruncate =
+    phoistAcyclic $
+        plam $ \ex x -> P.do
+            (PRational num denom) <- pmatch ex
+            mulDivTruncate # x # pto denom # num
+
+{- | Multiply a Rational by an Integer, without reducing the fraction
+
+ @since 3.8.0
+-}
+mulRational ::
+    forall (s :: S).
+    Term s (PInteger :--> PRational :--> PRational)
+mulRational =
+    phoistAcyclic $
+        plam $ \x r -> P.do
+            (PRational num denom) <- pmatch r
+            pcon $ PRational (num * x) denom
+
+{- | Multiply a Rational by an Integer, without reducing the fraction
+
+ @sinmce 3.8.0
+-}
```

Comment:
`since`

Also, weird argument order.

Reply:
Fixed the since.


RE argument order: I think the order of the comment is confusing (which I've fixed), but I think the actual order of the function makes sense. I think of things like `4 * (1/2)` or `4 / (1/2)` rather than the other way around.